src/include/wiredtiger.in

```diff
@@ -1897,6 +1897,24 @@ struct __wt_session {
 	int __F(timestamp_transaction)(WT_SESSION *session, const char *config);
 	/*! @} */
 
+#ifndef DOXYGEN
+	/*!
+	 * Set the commit timestamp on a transaction numerically.  Prefer this over @ref
+	 * timestamp_transaction when the string parsing done in that method becomes a bottleneck.
+	 *
+	 * The WT_SESSION.timestamp_transaction_commit method can only be used at snapshot
+	 * isolation.
+	 *
+	 * @requires_transaction
+	 *
+	 * @param session the session handle
+	 * @param ts the commit timestamp
+	 * @errors
+	 */
+	int __F(timestamp_transaction_commit)(WT_SESSION *session, unsigned long long ts);
```

Comment:
Maybe a better name will be "set_commit_timestamp" OR "set_commit_timestamp_numeric"

Reply:
I'm happy to rename it, but I'd like to get a little consensus first - every person I've spoken to has had a different idea so far ;-)

Reply:
I am also in favour of a name change. `set_numeric_timestamp`? And internal functions can be with their specific type like `set_numeric_commit_timestamp`?

Reply:
sure - it's probably best to discuss name changes on the ticket though, since more people are watching that. are you able to make this comment on the ticket please?